lms/djangoapps/student_account/views.py

```diff
@@ -83,7 +83,14 @@ def login_and_registration_form(request, initial_mode="login"):
         try:
             next_args = urlparse.parse_qs(urlparse.urlparse(redirect_to).query)
             provider_id = next_args['tpa_hint'][0]
-            if third_party_auth.provider.Registry.get(provider_id=provider_id):
+            tpa_hint_provider = third_party_auth.provider.Registry.get(provider_id=provider_id)
+            if tpa_hint_provider:
+                if tpa_hint_provider.skip_hinted_login_dialog:
+                    # Forward the user directly to the provider's login URL when the provider is configured
+                    # to skip the dialog.
+                    return redirect(
+                        pipeline.get_login_url(provider_id, pipeline.AUTH_ENTRY_LOGIN, redirect_url=redirect_to)
```

Comment:
Hmm, the presence of a valid `tpa_hint` always results in the `initial_mode` being overwritten to `hinted login` without this change (which then causes the view to show the hinted login prompt), so gating this new piece with logic based on `initial_mode` would just make it behave like it did before this change when users are navigating to the registration page with a `tpa_hint` (and prompt users for login instead).

I could maybe gate the whole `tpa_hint` block like this: `if initial_mode == "login" and '?' in redirect_to:`
which would make `tpa_hint` do nothing for the registration page. What do you think?